ic10emu/src/lib.rs

```diff
@@ -976,6 +978,35 @@ impl VM {
         }
     }
 
+    pub fn change_device_id(&mut self, old_id: u32, new_id: u32) -> Result<(), VMError> {
+        if self.devices.contains_key(&new_id) | self.ics.contains_key(&new_id) {
+            return Err(VMError::IdInUse(new_id));
+        }
+        let device = self
+            .devices
+            .remove(&old_id)
+            .ok_or(VMError::UnknownId(old_id))?;
+        device.borrow_mut().id = new_id;
+        self.devices.insert(new_id, device);
+        self.ics.iter().for_each(|(_id, ic)| {
+            if let Ok(mut ic_ref) = ic.try_borrow_mut() {
+                ic_ref.pins.iter_mut().for_each(|pin| {
+                    if pin.is_some_and(|d| d == old_id) {
+                        pin.replace(new_id);
+                    }
+                })
+            }
+        });
+        self.networks.iter().for_each(|(_net_id, net)| {
+            if let Ok(mut net_ref) = net.try_borrow_mut() {
```

Comment:
When would it already be borrowed? I guess same for pins, above, but failing to update pins isn't as bad as failing to update networks.